packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -0,0 +1,2966 @@
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+class RealmDictionaryTests : EmbeddedObjectCollectionQueryTests {
+
+    private val dictionarySchema = setOf(
+        RealmDictionaryContainer::class,
+        DictionaryLevel1::class,
+        DictionaryLevel2::class,
+        DictionaryLevel3::class,
+        DictionaryEmbeddedLevel1::class
+    )
+    private val descriptors = TypeDescriptor.allDictionaryFieldTypes
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    private val managedTesters: List<DictionaryApiTester<*, RealmDictionaryContainer>> by lazy {
+        descriptors.map {
+            val elementType = it.elementType
+            when (val classifier = elementType.classifier) {
+                RealmObject::class -> RealmObjectDictionaryTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as DictionaryTypeSafetyManager<RealmDictionaryContainer?>,
+                    classifier
+                )
+                ByteArray::class -> ByteArrayDictionaryTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as DictionaryTypeSafetyManager<ByteArray>,
+                    classifier
+                )
+                RealmAny::class -> RealmAnyDictionaryTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as DictionaryTypeSafetyManager<RealmAny?>,
+                    classifier
+                )
+                else -> GenericDictionaryTester(
+                    realm,
+                    getTypeSafety(classifier, elementType.nullable),
+                    classifier
+                )
+            }
+        }
+    }
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = dictionarySchema)
+            .directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun unmanaged() {
+        // No need to be exhaustive here, just checking delegation works
```

Comment:
Why don't we need to be exhaustive? We should be sure we know how unmanaged objects work as well. That said, It would not block merging this PR for it. Maybe just create an issue and we can add testing for it later.